### PR TITLE
Support importing json and fix builtins

### LINKED ISSUE #18: commonJS module (pouchdb-authentication) not importing correctly
I am writing an addon for adding [pouchdb-authentication](https://github.com/pouchdb-community/pouchdb-authentication) to ember-simple-auth. I am attempting to import pouchdb-authentication as a CommonJS module. 

I have done the following: 

```shell
npm install --save pouchdb-authentication
```

In my `index.js`, I have added the following hook: 
```javascript
included(app){
  app.import('node_modules/pouchdb-authentication/lib/index.js', {
    using: [
      {transform: 'cjs', as: 'pouchdb-authentication'}
    ]
  }
}
```

When I attempt to use the module in a class, ie `from PouchDBAuthentication import 'pouchdb-authentication`, I get the following error: 

```
Error: Could not find module `util` imported from `pouchdb-authentication`
at missingModule (http://localhost:7357/assets/vendor.js:252:11)
at findModule (http://localhost:7357/assets/vendor.js:263:7)
at Module.findDeps (http://localhost:7357/assets/vendor.js:173:24)
at findModule (http://localhost:7357/assets/vendor.js:267:11)
at Module.findDeps (http://localhost:7357/assets/vendor.js:173:24)
at findModule (http://localhost:7357/assets/vendor.js:267:11)
at requireModule (http://localhost:7357/assets/vendor.js:29:15)
at Class._extractDefaultExport (http://localhost:7357/assets/vendor.js:99921:20)
at Class.resolveOther (http://localhost:7357/assets/vendor.js:99621:32)
at Class.superWrapper [as resolveOther] (http://localhost:7357/assets/vendor.js:54195:22)
```

Any idea what is going on here? It does not look like `util` is an imported dependency in pouchdb-authentication.

### PR DESCRIPTION
**Pull request by Gaurav0**
This PR:

1. allows importing json files from within cjs files being imported via rollup-plugin-json
2. fixes issues with node builtins via rollup-plugin-node-builtins

Maybe fixes #18

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems good to me. @turbo87 - thoughts?

**Comment by Turbo87** [comment]
I'm a little worried about pulling in node builtins automatically without any explicit action. I've used browserify in the past that did this too and it resulted in the huge `Buffer` polyfill being included because something didn't work quite as planned. The `browser` property support of the node modules resolution should resolve that issue, but I'm still a little worried. I'm wondering if we should at least make it a configurable option for now until we've experiemented with it a little bit.

**Comment by Gaurav0** [comment]
@Turbo87 I am happy to make it configurable. What should the api for that be?

One possibility is:
```
app.import('node_modules/some-node-package/index.js', {
  using: [{ transformation: 'cjs', as: 'some-alias', builtins: true }]
});
```

**Comment by rwjblue** [comment]
Ya, we could also allow the importer to provide an arbitrary rollup plugins array to use instead...

**Comment by Gaurav0** [comment]
@rwjblue Like this?

```
const nodeBuiltins = require('rollup-plugin-node-builtins');
const json = require('rollup-plugin-json');
app.import('node_modules/some-node-package/index.js', {
  using: [{ transformation: 'cjs', as: 'some-alias', plugins: [nodeBuiltins, json] }]
});
```

Or this?
```
app.import('node_modules/some-node-package/index.js', {
  using: [{ transformation: 'cjs', as: 'some-alias', plugins: ['node-builtins',  'json'] }]
});
```

**Comment by rwjblue** [comment]
The former I believe

**Comment by Gaurav0** [comment]
@rwjblue @Turbo87 updated

**Comment by Gaurav0** [comment]
@rwjblue @Turbo87 Updated for merge conflicts. Could I please get some feedback?

**Comment by Turbo87** [comment]
@Gaurav0 do you think it's possible to add a test for this?

**Comment by Gaurav0** [comment]
@rwjblue @Turbo87 test added

**Comment by rwjblue** [comment]
Looks good to me, I think the only thing left is to add some small blurb of docs to the README around passing custom plugins.

**Comment by Gaurav0** [comment]
@rwjblue @Turbo87 Docs added.

@acorncom If you have a minute, can you look over my changes to README.md ?

File: README.md [hunk 1]
```diff
@@ -44,6 +44,26 @@ app.import('node_modules/showdown-highlight/lib/index.js', {
 app.import('node_modules/highlight.js/styles/tomorrow-night.css');
 ```
 
+There may be cases where you need additional rollup plugins in order to import your CommonJs files. For example, you may need to handle builtin node require statements, i.e.
+
+```
+var path = require('path');
+```
+
+For this, you may want to import the rollup plugin `rollup-plugin-node-builtins`. In order to do this, just include the plugin in your `package.json` and use the following syntax in your `ember-cli-build.js` or `index.js`:
```
Comment (sivakumar-kailasam) [inline README.md]: Let's not include 'just'.

File: README.md [hunk 2]
```diff
@@ -44,6 +44,26 @@ app.import('node_modules/showdown-highlight/lib/index.js', {
...
+const nodeBuiltins = require('rollup-plugin-node-builtins');
+
+app.import('node_modules/some-name/index.js', {
+  using: [
+    { transformation: 'cjs', as: 'some-name', plugins: [ nodeBuiltins() ]}
+  ]
+});
+```
+
+You can include multiple rollup plugins into the array and pass in options via the arguments to your plugin creation function.
```
Comment (sivakumar-kailasam) [inline README.md]: Shouldn't it be "you can include multiple plugins *in* the array"?

Comment (acorncom) [reply README.md]: Yep. @Gaurav0 with those two minor tweaks, looks good to me 👍

Comment (Gaurav0) [reply README.md]: Thanks!

**Review by sivakumar-kailasam — APPROVED** [review]

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you @Gaurav0!

**Pull request merged**